src/adapters/aof/local_aof.rs

```diff
@@ -0,0 +1,295 @@
+//! A local append-only file (AOF) adapter.
+
+use std::path::{Path, PathBuf};
+
+use anyhow::Result;
+
+use bytes::Bytes;
+use tokio::fs::{File, OpenOptions};
+use tokio::io::{AsyncReadExt, AsyncWriteExt, BufReader, BufWriter};
+
+use crate::services::interfaces::{AofEntry, TAof, WriteOperation};
+
+/// A local append-only file (AOF) implementation.
+pub struct LocalAof {
+    /// The file path where the AOF data is stored.
+    path: PathBuf,
+
+    /// A buffered writer for the underlying file.
+    writer: BufWriter<File>,
+
+    /// The highest offset successfully written or replayed.
+    current_offset: u64,
```

Comment:
What if AOF itself becomes an optional feature for performance reason? 

In that case entity responsible for managing offset would be delegated to something else?

Reply:
https://github.com/Migorithm/duva/pull/182#discussion_r1933346926